src/main/kotlin/com/pubnub/api/retry/RetryableRestCaller.kt

```diff
@@ -25,45 +22,51 @@ internal class RetryableRestCaller<T>(
     internal fun executeRestCallWithRetryPolicy(callToBeExecuted: Call<T>): Response<T> {
         call = callToBeExecuted
         var numberOfAttempts = 0
-        val maxRetryNumber = getRetryCount()
-        val maxAttempts = NUMBER_OF_REGULAR_CALLS + maxRetryNumber
         while (true) {
-            val response = executeRestCall()
-            if (!shouldRetry(response) || numberOfAttempts++ >= maxAttempts) {
-                return response
+            val (response, exception) = executeRestCall()
+            if (!shouldRetry(response) || numberOfAttempts++ >= maxRetryNumberFromConfiguration) {
+                if (response.isSuccessful || isHttpIssueRepresentedByErrorCodeNotByException(exception)) {
+                    return response
+                } else {
+                    throw exception!!
+                }
             }
-            val randomDelayInMilliSec = random.nextInt(BOUND)
-            val effectiveDelay = getDelayInMilliSeconds(response) + randomDelayInMilliSec
+            val randomDelayInMilliSec = random.nextInt(MAX_RANDOM_DELAY_IN_MILLI_SEC)
```

Comment:
I should have caught that in your previous comment - I believe "_MILLIS" is the usual way to say it (vs MILLI_SEC)

Reply:
ok. I will change that.